Runtime/Model/Metrics/MetricsSubmissionQueue.cs

```diff
@@ -116,8 +116,11 @@ internal void SendPayload(ICollection<T> events, uint attempts = 0)
                 {
                     OnRequestCompleted();
                 }
-                else if (statusCode > 501 && statusCode != 505)
+                else if (httpError == true || (statusCode > 501 && statusCode != 505))
                 {
+                    // Failed to communicate with server or received 4xx/5xx response code. Retry
```

Comment:
I think we also need to adjust the comment - we do not care about 4xx responses anymore, but we do care about networking issues or connection resets (or HTTP connection errors in general)

Reply:
Updated. Thank you.